text/2021-10-08-monitoring-metrics-about-chaos-mesh.md

```diff
@@ -0,0 +1,241 @@
+
+chaos-dashboard:
+
+- time histogram for each HTTP query
+- count for archived object
+- time histogram for archive reconciler
+
+## Detailed design
+
+### Metrics Plan
+
+The metrics design is as follows. It should be noted that the implementations
+of `chaos_controller_manager_chaos_experiments_count` and
+`chaos_daemon_grpc_server_handling_seconds` have been provided, and they are
+called `chaos_mesh_experiments` and `grpc_server_handling_seconds`. It is just
+hoping to modify the names to standardize naming. More about naming, see this
+guideline: [Metric and label naming](https://prometheus.io/docs/practices/naming/).
+
+<!-- markdownlint-disable MD013 -->
+| Name                                                  | Description                                            | Type         | Label                       | Buckets                      |
+|-------------------------------------------------------|--------------------------------------------------------|--------------|-----------------------------|------------------------------|
+| chaos_controller_manager_reconcile_duration_seconds   | time histogram for each `Reconcile()` in `Reconciler`  | HistogramVec | type                        | DefBuckets                   |
+| chaos_controller_manager_chaos_schedule_count         | currrent count of Schedule                             | GaugeVec     | namespace                   | /                            |
+| chaos_controller_manager_chaos_workflow_count         | currrent count of Workflow                             | GuageVec     | namespace                   | /                            |
+| chaos_controller_manager_emitted_event_count          | count of the emitted event by chaos-controller-manager | GaugeVec     | type, reason                | /                            |
+| chaos_controller_manager_grpc_client_handling_seconds | common metrics of grpc-client                          | HistogramVec | code, method, service, type | DefBuckets                   |
+| chaos_controller_manager_webhook_duration_seconds     | metrics for kubernetes webhook - duration              | HistogramVec | type, allowed               | WebhookDurationBuckets       |
+| chaos_controller_manager_webhook_request_total        | metrics for kubernetes webhook - request count         | CountVec     | type, allowed               | /                            |
+| chaos_daemon_grpc_server_handling_seconds             | common metrics of grpc-server                          | /            | /                           | ChaosDaemonGrpcServerBuckets |
+| chaos_daemon_bpm_controlled_process_total             | total of processes controlled by `bpm`                 | Count        | /                           | /                            |
+| chaos_daemon_bpm_controlled_process_count             | current count of processes controlled by `bpm`         | Guage        | /                           | /                            |
```

Comment:
Let's try to avoid suffix `count` for `Gauge` metrics. For example, this one can be renamed to `chaos_daemon_bpm_controlled_processes`.

Reply:
Okay, but can we use `number` as the suffix for `Gauge`? `chaos_daemon_bpm_controlled_processes` does not seem to indicate that this is a metric describing the quantity. How about `chaos_daemon_bpm_controlled_process_number`?

Reply:
I prefer to not use any suffix with Gauge.

Reply:
Okay.